### ISSUE TITLE
Doesn't work with Nextjs/SSR

### ISSUE THREAD

**Issue by matthewkeil** [mdtanrikulu/use-metamask#18]
Works fine with client render but server-side rendering errors because `window` is not defined

![Screen Shot 2021-11-11 at 12 55 25 AM](https://user-images.githubusercontent.com/18608739/141245510-349ff59a-a6b8-4dbe-ae39-8ebce6a77db1.png)

**Comment by guillermodlpa** [comment]
Hi all, thanks @matthewkeil for raising this issue, and for starting a PR https://github.com/mdtanrikulu/use-metamask/pull/19 to fix it! I came across it too.

As a temporary workaround until the fix is released, one can wrap the specific components using `useMetamask` with a no-SSR wrapper.

```jsx
<NoSsr>
  <ComponentLeveragingUseMetamask />
</NoSsr>
```

Here's Material-UI's implementation: https://github.com/mui-org/material-ui/blob/v5.2.6/packages/mui-base/src/NoSsr/NoSsr.js - Since I'm not using Material-UI for my project, I made a simpler component:

```jsx
const NoSsr = ({ children }: { children: React.ReactNode }) => {
  const [mountedState, setMountedState] = useState(false);
  useEffect(() => {
    setMountedState(true);
  }, []);
  return <>{mountedState ? children : null}</>;
};
```

Hope this helps until the fix is released! 

Also, @mdtanrikulu, thanks for creating `use-metamask`! It's so convenient!
